### PR TITLE
Fizzbuzz 2nd attempt

### PR DESCRIPTION
**Pull request by taha-crafter**
This is my 2nd attempt to do fizzbuzz according to the new grading checklist

### TODO

- [x] Each return value is a string 
- [x] 3 returns "Fizz"
- [x] 5 returns "Buzz
- [x] 15 returns "FizzBuzz"
- [x] 9 returns "Fizz"
- [x] 43 returns "43"
- [x] 42 returns "Fizz"
- [x] 45 returns "FizzBuzz"
- [x] 102 (you decide, throw an Error or handle some other way)
- [x] -12 (you decide, throw an Error or handle some other way)
- [x] any non-number (you decide, throw an Error or handle some other way)

### CODE REVIEW

File: 1_The_Physical/Guess_Point_Level_1_Stateless/1_Red_Green_Refactor/1_1_Fizzbuzz/src/fizzbuzz.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 export const fizzbuzz = (num: number): string => {
   if (num > 100) throw new Error('Number must be less than 100');
+  if (num < 0) throw new Error('Number must be greater than 0');
```
Comment (n0rdlicht) [inline 1_The_Physical/Guess_Point_Level_1_Stateless/1_Red_Green_Refactor/1_1_Fizzbuzz/src/fizzbuzz.ts:4]: **question**: The check might needs to be `num <= 0` or `num < 1` as the number should be greater then 0.

File: 1_The_Physical/Guess_Point_Level_1_Stateless/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts [hunk 2]
```diff
@@ -1,5 +1,45 @@
+import { fizzbuzz } from './fizzbuzz';
 
-describe("fizzbuzz", () => {
+describe('fizzbuzz', () => {
+  describe('Accept only input of type number, between 1 and 100, and returns string', () => {
+    it('throws an error when given non-number input', () => {
+      expect(() => fizzbuzz('hello' as any)).toThrow();
```
Comment (n0rdlicht) [inline 1_The_Physical/Guess_Point_Level_1_Stateless/1_Red_Green_Refactor/1_1_Fizzbuzz/src/index.spec.ts:6]: **question(non-blocking)**: We might be testing typescript here, even though its in the checklist I'm not sure it makes sense to test and hence implement this one...

**Review by n0rdlicht — COMMENTED** [review]
I've added two smaller questions, hope they are useful.

File: 1_The_Physical/Guess_Point_Level_1_Stateless/1_Red_Green_Refactor/1_1_Fizzbuzz/src/fizzbuzz.ts (hunk 1 above)
Comment (taha-programmer) [reply 1_The_Physical/Guess_Point_Level_1_Stateless/1_Red_Green_Refactor/1_1_Fizzbuzz/src/fizzbuzz.ts:4]: Yes, Thanks!
It accepts numbers between 1 and 100, so it should be < 1

**Pull request closed without merging**